### PR TITLE
job-exec: fix potential leak of job KVS namespaces

### PR DESCRIPTION
**Pull request by grondo**
This PR fixes the issue described in #5790. If a job is canceled or gets a fatal job exception while the namespace create RPC is in flight, the fact that a namespace exists for the job is dropped and the namespace will be leaked.

Also included in this PR is an improvement to the `exec_kill` error message handling to reduce a non-trivial amount of log noise for large jobs.

### CODE REVIEW

File: src/modules/job-exec/job-exec.c [hunk 1]
```diff
@@ -696,7 +696,8 @@ static flux_future_t * namespace_move (struct jobinfo *job)
 {
```
Comment (garlick) [inline src/modules/job-exec/job-exec.c:696]: In first line of commit description in `fix potential namespace leak`, there is a stray `a`.

**Review by garlick — APPROVED** [review]
LGTM!

File: src/modules/job-exec/job-exec.c (hunk 1 above)
Comment (grondo) [reply src/modules/job-exec/job-exec.c:696]: Thanks! Fixed and will set MWP.

**Comment by grondo** [comment]
Hm, the distcheck build failed here:
```
 expecting success: 
  	jobid=$(flux submit --wait-event=exec.shell.init sleep inf)
  	watchers=$(get_update_watchers)
  	${UPDATE_WATCH} $jobid R > watch10A.out &
  	watchpidA=$! &&
  	wait_update_watchers $((watchers+1)) &&
  	update1=$(expiration_add $jobid 100)
  	${UPDATE_WATCH} $jobid R > watch10B.out &
  	watchpidB=$! &&
  	wait_update_watchers $((watchers+2)) &&
  	${WAITFILE} --count=2 --timeout=30 --pattern="expiration" watch10A.out &&
  	${WAITFILE} --count=1 --timeout=30 --pattern="expiration" watch10B.out &&
  	kill -s USR1 $watchpidA &&
  	wait $watchpidA &&
  	kill -s USR1 $watchpidB &&
  	wait $watchpidB &&
  	test_debug "echo watch10A: \"$(cat watch10A.out)\"" &&
  	test_debug "echo watch10B: \"$(cat watch10B.out)\"" &&
  	flux cancel $jobid &&
  	test $(cat watch10A.out | wc -l) -eq 2 &&
  	test $(cat watch10B.out | wc -l) -eq 1 &&
  	head -n1 watch10A.out | jq -e ".execution.expiration == 0.0" &&
  	tail -n1 watch10B.out | jq -e ".execution.expiration == ${update1}" &&
  	cat watch10B.out | jq -e ".execution.expiration == ${update1}"
  
  waiting for 2 watchers
  2
  waiting for 3 watchers
  3
  {"version": 1, "execution": {"R_lite": [{"rank": "0", "children": {"core": "0"}}], "starttime": 1710864460.7675567, "expiration": 0.0, "nodelist": ["fv-az528-180"]}}
  {"version": 1, "execution": {"R_lite": [{"rank": "0", "children": {"core": "0"}}], "starttime": 1710864460.7675567, "expiration": 1710864560, "nodelist": ["fv-az528-180"]}}
  {"version": 1, "execution": {"R_lite": [{"rank": "0", "children": {"core": "0"}}], "starttime": 1710864460.7675567, "expiration": 0.0, "nodelist": ["fv-az528-180"]}}
  not ok 13 - job-info: update watch can be canceled (multiple watchers)
```

**Pull request merged**